tests/acceptance/redirect-on-permission-test.js

```diff
@@ -0,0 +1,508 @@
+import Ember from 'ember';
+import { module, test, skip } from 'qunit';
+import startApp from '../helpers/start-app';
+import { stubRequest } from 'ember-cli-fake-server';
+import { orgId, rolesHref, usersHref, invitationsHref,
+         securityOfficerHref } from '../helpers/organization-stub';
+
+let application;
+let userId = 'user1';
+let stackId = 'my-stack-1';
+let enclaveUrl = `/stacks/${stackId}`;
+let gridironAdminUrl = `/gridiron/${orgId}/admin`;
+let myGridironUrl = `/gridiron/${orgId}/user`;
+let organizationAdminUrl = `/organizations/${orgId}/admin/contact-settings`;
+let users = [
+  {
+    id: userId,
+    name: 'Basic User',
+    email: 'basicuser@asdf.com',
+    _links: {
+      self: { href: `/users/${userId}` }
+    }
+  }
+];
+
+module('Acceptance: Product Dashboard Permissions', {
+  beforeEach() {
+    application = startApp();
+    stubProfile({ hasCompletedSetup: true });
+    stubCriterionDocuments({});
+    stubCriteria();
+    stubStacks();
+    stubRequest('get', '/documents', function() {
+      return this.success({ _embedded: { documents: [] }});
+    });
+  },
+
+  afterEach() {
+    Ember.run(application, 'destroy');
+  }
+});
+
+test('Users in `Owners` Role', function(assert) {
```

Comment:
The primary purpose of this PR was to make these test cases as easy as possible to detect and honor in dashboard.  Most other changes just precipitated from being able to support all of these use cases.